### PR TITLE
Jenkins release job

### PR DESCRIPTION
**Pull request by sbesson**
This PR adds necessary changes to setup an automatic OME-release job on Jenkins performing the following:
- create new release folders 
- symlink the promoted OMERO & BIOFORMATS artifacts
- generate downloads pages using `gen.py` and `bfgen.py`

### CODE REVIEW

**Comment by sbesson** [comment]
See [OME-merge-release](http://hudson.openmicroscopy.org.uk/job/OME-merge-release/) for the Jenkins implementation of these changes.

File: gen.py [hunk 1]
```diff
@@ -21,7 +21,7 @@
 fingerprint_url = "http://hudson.openmicroscopy.org.uk/fingerprint"
 MD5s = """
 MD5(4.4.5/OMERO-4.4.5.pdf)= d92d4bd40e2defb328756780c77a633e
-MD5(4.4.6/OMERO-4.4.6.pdf)= f18208f6c0a53083392dbeb590310730
+MD5(4.4.6/OMERO-4.4.6.pdf)= 03aace065d939bb7a47a13b47cf4cc70
```
Comment (joshmoore) [inline gen.py:24]: Hmmm.......this might become a problem, no? Maybe keep both?

**Comment by joshmoore** [comment]
Job looks amazing. Since this is a new repository, there's not really much danger in merging. Are there any next steps you want on this PR?

File: gen.py (hunk 1 above)
Comment (sbesson) [inline gen.py:24]: Pdf should now be archived in the OMERO & BF jobs (and copied over during promotion). Ideally I want to get rid of these hard-coded lines.

**Comment by sbesson** [comment]
Couple of immediate things I want to do:
- Allow to pass `SNAPSHOT_URL` by envvar in `bfgen.py`, `gen.py` to test links with test job
- apply the job to most recent promoted builds (4.4.7-DEV or 5.0.0-alpha1). Should certainly involve modifications (since OMERO/OMERO-ice34 jobs are different)
  I suppose the 2d can wait for another dedicated PR. Maybe I can work on the first item in this PR?

**Comment by joshmoore** [comment]
Is it perhaps time to make `OMERO-{stable|develop}-ice3{34}-merge` which just build as the release jobs do and make the release job depend on those?

**Comment by sbesson** [comment]
@joshmoore: not sure I fully get it, what do you mean by `OMERO-{stable|develop}-ice3{34}-merge` building as a release job?

**Comment by sbesson** [comment]
See [OME-merge-release#41](http://hudson.openmicroscopy.org.uk/view/Mgmt/job/OME-merge-release/41/)

**Comment by joshmoore** [comment]
I mean that we have a job identical to OMERO-stable and OMERO-stable-ice34 which instead of just building `dev_4_4` first does a SCC-merge. And the same for trunk/develop.

**Comment by joshmoore** [comment]
The job is certainly happy, so I guess I am too. Merging.

**Pull request merged**